config/sampler_config.go

```diff
@@ -45,12 +47,379 @@ type RulesBasedSamplerCondition struct {
 	Field    string
 	Operator string
 	Value    interface{}
+	Datatype string
+	Matches  func(value any, exists bool) bool
+}
+
+func (r *RulesBasedSamplerCondition) Init() error {
+	return r.setMatchesFunction()
 }
 
 func (r *RulesBasedSamplerCondition) String() string {
 	return fmt.Sprintf("%+v", *r)
 }
 
+func (r *RulesBasedSamplerCondition) setMatchesFunction() error {
+	switch r.Operator {
+	case "exists":
+		r.Matches = func(value any, exists bool) bool {
+			return exists
+		}
+	case "not-exists":
+		r.Matches = func(value any, exists bool) bool {
+			return !exists
+		}
+	case "!=":
+		return setCompareOperators(r, "!=")
+	case "=":
+		return setCompareOperators(r, "=")
+	case ">":
+		return setCompareOperators(r, ">")
+	case ">=":
+		return setCompareOperators(r, ">=")
+	case "<":
+		return setCompareOperators(r, "<")
+	case "<=":
+		return setCompareOperators(r, "<=") //which format is better?
+	case "starts-with":
+		err := setMatchStringBasedOperators(r, "starts-with")
+		if err != nil {
+			return err
+		}
+	case "contains":
+		err := setMatchStringBasedOperators(r, "contains")
+		if err != nil {
+			return err
+		}
+	case "does-not-contain":
+		err := setMatchStringBasedOperators(r, "does-not-contain")
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown operator '%s'", r.Operator)
+	}
+	return nil
```

Comment:
nit: first time reading through I got confused how this return could be hit but then I saw that `exists` and `not exists` don't return.  Since we have a default case I'd opt for `exists` and `not exists` returning nil so that the default case is the guaranteed return statement.

Reply:
True, this is better.